webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeGroupRepository.scala

```diff
@@ -489,44 +497,44 @@ class WoLDAPNodeGroupRepository(
    * Check if a group category exist with the given name
    */
   private[this] def categoryExists(con:RoLDAPConnection, name : String, parentDn : DN) : IOResult[Boolean] = {
-    con.searchOne(parentDn, AND(IS(OC_GROUP_CATEGORY), EQ(A_NAME, name)), A_GROUP_CATEGORY_UUID).flatMap(_.size match {
+    groupLibMutex.readLock(con.searchOne(parentDn, AND(IS(OC_GROUP_CATEGORY), EQ(A_NAME, name)), A_GROUP_CATEGORY_UUID).flatMap(_.size match {
```

Comment:
the categoryExists are used only in this class, and within writeLock. I'm unsure we need a readLock there

Reply:
We don't really need it yet, but I wanted to tag all access to `con` with a corresponding `read`/`write` lock level, in case of refactoring or code evolution.